File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Beginners/Selenium/Bing/Pages/BingMainPage.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace PatternsInAutomatedTests.Beginners.Selenium.Bing.Pages
{
    public class BingMainPage
    {
        private readonly IWebDriver driver;
        private readonly string url = @"http://www.bing.com/";

        public BingMainPage(IWebDriver browser)
        {
            this.driver = browser;
            PageFactory.InitElements(browser, this);
        }

        [FindsBy(How = How.Id, Using = "sb_form_q")]
        public IWebElement SearchBox { get; set; }

        [FindsBy(How = How.Id, Using = "sb_form_go")]
        public IWebElement GoButton { get; set; }

        [FindsBy(How = How.Id, Using = "b_tween")]
        public IWebElement ResultsCountDiv { get; set; }

        public void Navigate()
        {
            this.driver.Navigate().GoToUrl(this.url);            
        }

        public void Search(string textToType)
        {
            this.SearchBox.Clear();
            this.SearchBox.SendKeys(textToType);
            this.GoButton.Click();
        }

        public void ValidateResultsCount(string expectedCount)
        {
            Assert.IsTrue(this.ResultsCountDiv.Text.Contains(expectedCount), "The results DIV doesn't contains the specified text.");
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/com/automatetheplanet/_1_page_object_pattern/Selenium/Bing/Pages/BingMainPage.java
package com.automatetheplanet._1_page_object_pattern.Selenium.Bing.Pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.testng.Assert;

public class BingMainPage {
    private final WebDriver driver;
    private final String url = "http://www.bing.com/";

    public BingMainPage(WebDriver browser) {
        driver = browser;
        PageFactory.initElements(browser, this);
    }

    @FindBy(id = "sb_form_q")
    public WebElement searchBox;

    @FindBy(xpath = "//label[@for='sb_form_go']")
    public WebElement goButton;

    @FindBy(id = "b_tween")
    public WebElement resultsCountDiv;

    public void navigate() {
        driver.navigate().to(url);
    }

    public void search(String textToType) {
        searchBox.clear();
        searchBox.sendKeys(textToType);
        goButton.click();
    }

    public void assertsResultsCount(String expectedCount) {
        Assert.assertTrue(resultsCountDiv.getText().contains(expectedCount), "The results DIV doesn't contain the specified text.");
    }
}
